src/bin/dhcp4/dhcp4_srv.cc

```diff
@@ -1191,10 +1191,22 @@ Dhcpv4Srv::assignLease(Dhcpv4Exchange& ex) {
         // misconfiguration of the server (or some relays).
 
         // Perhaps this should be logged on some higher level?
-        LOG_ERROR(bad_packet4_logger, DHCP4_PACKET_NAK_0001)
-            .arg(query->getLabel())
-            .arg(query->getRemoteAddr().toText())
-            .arg(query->getName());
+        if (HooksManager::calloutsPresent(hook_index_pkt4_send_)) {
```

Comment:
the reason I am doing this is that since we don't use the lease assignment functionality in our deployment the ERROR log lines are spammy.